From b6cc7051ecbfe564e229a4d463b43711f5370efa Mon Sep 17 00:00:00 2001
From: James Elliott <james.elliott@acquia.com>
Date: Thu, 12 May 2011 14:02:20 -0400
Subject: [PATCH] Fix multi value fields with WYSIWYG

---
 wysiwyg.js |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/wysiwyg.js b/wysiwyg.js
index 4225f09..6ca749f 100644
--- a/wysiwyg.js
+++ b/wysiwyg.js
@@ -80,6 +80,13 @@ Drupal.behaviors.attachWysiwyg = {
         Drupal.wysiwygDetach(context, params[format]);
       });
     });
+  },
+  detach: function(context, settings) {
+    $('.wysiwyg-processed', context).each(function (index, element) {
+      $(element).removeClass('wysiwyg-processed');
+      var params = Drupal.settings.wysiwyg.triggers[element.id];
+      Drupal.wysiwygDetach(context, params);
+    });
   }
 };
 
-- 
1.7.4.msysgit.0

