diff --git a/js/views_bulk_operations.js b/js/views_bulk_operations.js
index aeeecc6..8811592 100644
--- a/js/views_bulk_operations.js
+++ b/js/views_bulk_operations.js
@@ -32,7 +32,7 @@
     // This is the "select all" checkbox in (each) table header.
     $('.vbo-table-select-all', form).click(function() {
       var table = $(this).closest('table')[0];
-      $('input[id^="edit-views-bulk-operations"]:not(:disabled)', table).attr('checked', this.checked);
+      $('input[id^="edit-views-bulk-operations"]:not(:disabled)', table).attr('checked', this.checked).change();
 
       // Toggle the visibility of the "select all" row (if any).
       if (this.checked) {
@@ -49,15 +49,8 @@
     $('.views-table tbody tr', form).click(function(event) {
       if (event.target.tagName.toLowerCase() != 'input' && event.target.tagName.toLowerCase() != 'a') {
         $('input[id^="edit-views-bulk-operations"]:not(:disabled)', this).each(function() {
-          var checked = this.checked;
-          // trigger() toggles the checkmark *after* the event is set,
-          // whereas manually clicking the checkbox toggles it *beforehand*.
-          // that's why we manually set the checkmark first, then trigger the
-          // event (so that listeners get notified), then re-set the checkmark
-          // which the trigger will have toggled. yuck!
-          this.checked = !checked;
-          $(this).trigger('click');
-          this.checked = !checked;
+          this.checked = !this.checked;
+          $(this).change();
         });
       }
     });
@@ -81,35 +74,35 @@
     $('.vbo-select-all-markup', form).show();
 
     $('.vbo-select-this-page', form).click(function() {
-      $('input[id^="edit-views-bulk-operations"]', form).attr('checked', this.checked);
-      $('.vbo-select-all-pages', form).attr('checked', false);
+      $('input[id^="edit-views-bulk-operations"]', form).attr('checked', this.checked).change();
+      $('.vbo-select-all-pages', form).attr('checked', false).change();
 
       // Toggle the "select all" checkbox in grouped tables (if any).
-      $('.vbo-table-select-all', form).attr('checked', this.checked);
+      $('.vbo-table-select-all', form).attr('checked', this.checked).change();
     });
     $('.vbo-select-all-pages', form).click(function() {
-      $('input[id^="edit-views-bulk-operations"]', form).attr('checked', this.checked);
-      $('.vbo-select-this-page', form).attr('checked', false);
+      $('input[id^="edit-views-bulk-operations"]', form).attr('checked', this.checked).change();
+      $('.vbo-select-this-page', form).attr('checked', false).change();
 
       // Toggle the "select all" checkbox in grouped tables (if any).
-      $('.vbo-table-select-all', form).attr('checked', this.checked);
+      $('.vbo-table-select-all', form).attr('checked', this.checked).change();
 
       // Modify the value of the hidden form field.
       $('.select-all-rows', form).val(this.checked);
     });
 
-    $('.vbo-select', form).click(function() {
+    $('.vbo-select', form).change(function() {
       // If a checkbox was deselected, uncheck any "select all" checkboxes.
       if (!this.checked) {
-        $('.vbo-select-this-page', form).attr('checked', false);
-        $('.vbo-select-all-pages', form).attr('checked', false);
+        $('.vbo-select-this-page', form).attr('checked', false).change();
+        $('.vbo-select-all-pages', form).attr('checked', false).change();
         // Modify the value of the hidden form field.
         $('.select-all-rows', form).val('0')
 
         var table = $(this).closest('table')[0];
         if (table) {
           // Uncheck the "select all" checkbox in the table header.
-          $('.vbo-table-select-all', table).attr('checked', false);
+          $('.vbo-table-select-all', table).attr('checked', false).change();
 
           // If there's a "select all" row, hide it.
           if ($('.vbo-table-select-this-page', table).length) {
