diff --git a/views_send.js b/views_send.js
index 1401db0..9d510a0 100644
--- a/views_send.js
+++ b/views_send.js
@@ -14,7 +14,7 @@
     // This is the "select all" checkbox in (each) table header.
     $('.views-send-table-select-all', form).click(function() {
       var table = $(this).closest('table')[0];
-      $('input[id^="edit-views-send"]:not(:disabled)', table).attr('checked', this.checked);
+      $('input[id^="edit-views-send"]:not(:disabled)', table).attr('checked', this.checked).change();
     });
   }
 
@@ -23,21 +23,21 @@
     $('.views-send-select-all-markup', form).show();
 
     $('.views-send-select-this-page', form).click(function() {
-      $('input[id^="edit-views-send"]', form).attr('checked', this.checked);
+      $('input[id^="edit-views-send"]', form).attr('checked', this.checked).change();
 
       // Toggle the "select all" checkbox in grouped tables (if any).
-      $('.views-send-table-select-all', form).attr('checked', this.checked);
+      $('.views-send-table-select-all', form).attr('checked', this.checked).change();
     });
 
     $('.views-send-select', form).click(function() {
       // If a checkbox was deselected, uncheck any "select all" checkboxes.
       if (!this.checked) {
-        $('.views-send-select-this-page', form).attr('checked', false);
+        $('.views-send-select-this-page', form).attr('checked', false).change();
 
         var table = $(this).closest('table')[0];
         if (table) {
           // Uncheck the "select all" checkbox in the table header.
-          $('.views-send-table-select-all', table).attr('checked', false);
+          $('.views-send-table-select-all', table).attr('checked', false).change();
         }
       }
     });
