diff --git a/modules/user/user.pages.inc b/modules/user/user.pages.inc
index f21bd13..7c09263 100644
--- a/modules/user/user.pages.inc
+++ b/modules/user/user.pages.inc
@@ -73,11 +73,8 @@ function user_pass_validate($form, &$form_state) {
 }

 function user_pass_submit($form, &$form_state) {
-  global $language;
-
   $account = $form_state['values']['account'];
-  // Mail one time login URL and instructions using current language.
-  $mail = _user_mail_notify('password_reset', $account, $language);
+  $mail = _user_mail_notify('password_reset', $account);
   if (!empty($mail)) {
     watchdog('user', 'Password reset instructions mailed to %name at %email.', array('%name' => $account->name, '%email' => $account->mail));
     drupal_set_message(t('Further instructions have been sent to your e-mail address.'));
