diff --git a/logentries.install b/logentries.install
index a9d993c..70cd2cf 100644
--- a/logentries.install
+++ b/logentries.install
@@ -70,5 +70,6 @@ function logentries_uninstall() {
   variable_del('logentries_ssl');
   variable_del('logentries_severity');
   variable_del('logentries_strip_html');
+  variable_del('logentries_port');
 }

diff --git a/logentries.module b/logentries.module
index b3adffe..5246a1c 100644
--- a/logentries.module
+++ b/logentries.module
@@ -161,7 +161,7 @@ function logentries_init_log() {
         $severity = variable_get('logentries_severity', WATCHDOG_DEBUG);

         if (logentries_count_getlogger_params() > 4) {
-          $_logentries_log = LeLogger::getLogger($token, $persistent, $ssl, $severity, FALSE, '', 10000, '', '', FALSE);
+          $_logentries_log = LeLogger::getLogger($token, $persistent, $ssl, $severity, FALSE, '', logentries_get_port_number(), '', '', FALSE, FALSE);
         }
         else {
           $_logentries_log = LeLogger::getLogger($token, $persistent, $ssl, $severity);
@@ -227,6 +227,22 @@ function logentries_form_system_logging_settings_alter(&$form, &$form_state, $fo
       '#title' => t('SSL'),
       '#type' => 'checkbox',
     );
+    $form['logentries']['logentries_port'] = array(
+      '#default_value' => variable_get('logentries_port', 80),
+      '#description' => t('Select the port number to use when sending log messages via TCP or UDP.'),
+      '#title' => t('Port'),
+      '#type' => 'radios',
+      '#options' => array(
+        80 => 80,
+        514 => 514,
+        10000 => 10000,
+      ),
+      '#states' => array(
+        'visible' => array(
+          array('[name=logentries_ssl]' => array('checked' => FALSE)),
+        ),
+      ),
+    );
     $form['logentries']['logentries_strip_html'] = array(
       '#default_value' => variable_get('logentries_strip_html', TRUE),
       '#description' => t('Strip HTML tags from log messages.'),
@@ -288,9 +304,10 @@ function logentries_form_system_logging_settings_alter_test_submit($form, &$form
   $persistent = TRUE;
   $ssl = $form_state['values']['logentries_ssl'];
   $severity = $form_state['values']['logentries_severity'];
+  $port = $ssl ? 20000 : $form_state['values']['logentries_port'];

   if (logentries_count_getlogger_params() > 4) {
-    $log = LeLogger::getLogger($token, $persistent, $ssl, $severity, FALSE, '', 10000, '', '', FALSE);
+    $log = LeLogger::getLogger($token, $persistent, $ssl, $severity, FALSE, '', $port, '', '', FALSE, FALSE);
   }
   else {
     $log = LeLogger::getLogger($token, $persistent, $ssl, $severity);
@@ -315,6 +332,24 @@ function logentries_form_system_logging_settings_alter_test_submit($form, &$form
 }

 /**
+ * Get the port number to use.
+ *
+ * This will return the configured port number if SSL has been turned of. If SSL
+ * is enabled, we'll force the port to 20000 since that's the only one supported
+ * by Logentries.
+ *
+ * @return int
+ *   The port number.
+ */
+function logentries_get_port_number() {
+  if (variable_get('logentries_ssl', FALSE)) {
+    return 20000;
+  }
+
+  return variable_get('logentries_port', 80);
+}
+
+/**
  * Check the number of parameters expected by the getLogger method.
  *
  * This is to check which version of the getLogger method is being used, to
