Index: l10n_update.module
===================================================================
--- l10n_update.module	(revision 162)
+++ l10n_update.module	(working copy)
@@ -310,7 +310,7 @@
 function l10n_update_get_projects($refresh = FALSE, $disabled = FALSE) {
   static $projects, $enabled;
 
-  if (!isset($projects) || $refresh) {
+  if (!isset($projects) || $refresh || drupal_installation_attempted()) {
     if (variable_get('l10n_update_rebuild_projects', 0)) {
       module_load_include('project.inc', 'l10n_update');
       variable_del('l10n_update_rebuild_projects');
