diff --git a/handlers/views_handler_filter.inc b/handlers/views_handler_filter.inc
index b8c6ac1..3bb4e85 100644
--- a/handlers/views_handler_filter.inc
+++ b/handlers/views_handler_filter.inc
@@ -455,7 +455,18 @@ class views_handler_filter extends views_handler {
     // Build the form and set the value based on the identifier.
     if (!empty($this->options['expose']['identifier'])) {
       $value = $this->options['expose']['identifier'];
-      $this->value_form($form, $form_state);
+      if ($this->operator == 'empty' || $this->operator == 'not empty') {
+        $this->value = 'All';
+        $this->value_value = '';
+        $this->value_options = array(
+          1 => t('Yes'),
+          0 => t('No')
+        );
+        views_handler_filter_boolean_operator::value_form($form, $form_state);
+      }
+      else {
+        $this->value_form($form, $form_state);
+      }
       $form[$value] = $form['value'];
 
       if (isset($form[$value]['#title']) && !empty($form[$value]['#type']) && $form[$value]['#type'] != 'checkbox') {
@@ -585,14 +596,13 @@ class views_handler_filter extends views_handler {
 
       // Various ways to check for the absence of non-required input.
       if (empty($this->options['expose']['required'])) {
-        if (($this->operator == 'empty' || $this->operator == 'not empty') && $value === '') {
-          $value = ' ';
+        if ($this->operator == 'empty' || $this->operator == 'not empty') {
+          $value = is_array($value) ? $value['value'] : $value;
+          $this->operator = ($this->operator == 'empty' && empty($value)) || ($this->operator == 'not empty' && !empty($value)) ? 'not empty' : 'empty';
         }
 
-        if ($this->operator != 'empty' && $this->operator != 'not empty') {
-          if ($value == 'All' || $value === array()) {
-            return FALSE;
-          }
+        if ($value == 'All' || $value === array()) {
+          return FALSE;
         }
 
         if (!empty($this->always_multiple) && $value === '') {
