Index: includes/bootstrap.inc
===================================================================
RCS file: /cvs/drupal/drupal/includes/bootstrap.inc,v
retrieving revision 1.206.2.11
diff -u -p -r1.206.2.11 bootstrap.inc
--- includes/bootstrap.inc	25 Feb 2009 13:49:54 -0000	1.206.2.11
+++ includes/bootstrap.inc	8 May 2009 01:02:09 -0000
@@ -629,7 +629,7 @@ function drupal_load($type, $name) {
 function drupal_page_header() {
   header("Expires: Sun, 19 Nov 1978 05:00:00 GMT");
   header("Last-Modified: ". gmdate("D, d M Y H:i:s") ." GMT");
-  header("Cache-Control: store, no-cache, must-revalidate");
+  header("Cache-Control: no-store, no-cache, must-revalidate");
   header("Cache-Control: post-check=0, pre-check=0", FALSE);
 }
 
@@ -666,7 +666,7 @@ function drupal_page_cache_header($cache
 
   // The following headers force validation of cache:
   header("Expires: Sun, 19 Nov 1978 05:00:00 GMT");
-  header("Cache-Control: must-revalidate");
+  header("Cache-Control: no-store, no-cache, must-revalidate");
 
   if (variable_get('page_compression', TRUE)) {
     // Determine if the browser accepts gzipped data.
