diff --git a/views_handler_area_link.inc b/views_handler_area_link.inc
index c784695..f45cddd 100644
--- a/views_handler_area_link.inc
+++ b/views_handler_area_link.inc
@@ -86,11 +86,9 @@ class views_handler_area_link extends views_handler_area {
       // Replace any tokens in the path.
       $path = strtr($this->options['path'], $tokens);
 
-      // Check that the user has access to the menu router item, but only if the
-      // path is for a valid menu router item, so that external URLs or paths not
-      // handled by Drupal's menu router are always permitted.
-      $router_item = menu_get_item($path);
-      if ($router_item && !$router_item['access']) {
+      // Check that path is valid and user has access to it.
+      // External paths are fine.
+      if (!drupal_valid_path($path)) {
         return '';
       }
 
